dubbo-config/dubbo-config-spring/src/test/java/org/apache/dubbo/config/spring/beans/factory/annotation/DubboReferencePostProcessAfterInitializationTest.java

```diff
@@ -0,0 +1,96 @@
+
+import java.lang.reflect.Proxy;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.dubbo.config.annotation.DubboReference;
+import org.apache.dubbo.config.spring.ReferenceBean;
+import org.apache.dubbo.config.spring.api.HelloService;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import static org.apache.dubbo.config.spring.beans.factory.annotation.ReferenceAnnotationBeanPostProcessor.BEAN_NAME;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * {@link DubboReference @DubboReference} of Generic injection test
+ *
+ * @see DubboReference
+ * @since 2.7.9
+ */
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(
+        classes = {
+                ServiceAnnotationTestConfiguration.class,
+                DubboReferencePostProcessAfterInitializationTest.class
+        })
+@TestPropertySource(properties = {
+        "packagesToScan = org.apache.dubbo.config.spring.context.annotation.provider",
+        "consumer.version = ${demo.service.version}",
+        "consumer.url = dubbo://127.0.0.1:12345?version=2.5.7",
+})
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+public class DubboReferencePostProcessAfterInitializationTest {
+
+    private static final Map<String, ReferenceBean<?>> REFERENCE_BEAN_MAP = new HashMap<>(1, 1);
+
+    @Bean(BEAN_NAME)
+    public ReferenceAnnotationBeanPostProcessor referenceAnnotationBeanPostProcessor() {
+        return new ReferenceAnnotationBeanPostProcessor();
+    }
+
+    @DubboReference
+    private HelloService helloService;
+
+    @Test
+    public void test() {
+        assertEquals(1, REFERENCE_BEAN_MAP.size());
+        assertEquals("Greeting, Mercy.", helloService.sayHello("Mercy"));
+    }
+
+    @Bean("afterBeanPostProcessor")
+    public AfterBeanPostProcessor afterBeanPostProcessor() {
+        return new AfterBeanPostProcessor();
+    }
+
+    public static class AfterBeanPostProcessor implements BeanPostProcessor {
+
+        public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
+            if (bean.getClass() == ReferenceBean.class && ((ReferenceBean<?>) bean).getInterfaceClass() == HelloService.class) {
+                REFERENCE_BEAN_MAP.put(beanName, (ReferenceBean<?>) bean);
+            } else if (REFERENCE_BEAN_MAP.containsKey(beanName)) {
+                return Proxy.newProxyInstance(getClass().getClassLoader(), new Class<?>[]{HelloService.class}, (proxy, method, args) -> {
```

Comment:
This test case is a bit weird. Why use a custom BeanPostProcessor to create a wrapper proxy? It would be better if you use Aspect to do it. 
I plan to pick this test case  to the 3.0 branch.

Reply:
Test scenarios such as the reference bean generated by dubbroeference support calling applybean postprocessor after initialization, and the instance generated by this reference bean supports calling applybean postprocessor after initialization